pkg/qsfsd/qsfs.go

```diff
@@ -100,8 +103,50 @@ func (q *QSFS) Mount(wlID string, cfg zos.QuatumSafeFS) (mountPath string, err e
 		qsfsContainerNS,
 		cont,
 	)
+	if lerr := q.waitUntilMounted(mountPath, 2*time.Minute); lerr != nil {
+		err = lerr
+		return
+	}
 	return
 }
+
+func (f *QSFS) waitUntilMounted(path string, timeout time.Duration) error {
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(1 * time.Second) {
+		mounted, err := f.isMounted(path)
+		if err != nil {
+			return errors.Wrap(err, "failed to check if zdbfs is mounted")
+		}
+		if mounted {
+			return nil
+		}
+	}
+	return fmt.Errorf("waiting for zdbfs mount %s timedout", path)
+
+}
+
+func (f *QSFS) isMounted(path string) (bool, error) {
+	output, err := exec.Command("findmnt", "-J", path).Output()
+	if err, ok := err.(*exec.ExitError); ok && err != nil {
+		if err.ExitCode() == 1 {
+			return false, nil
+		}
+	}
+	var result struct {
+		Filesystems []struct {
+			Fstype string `json:"fstype"`
+		} `json:"filesystems"`
+	}
+
+	if err := json.Unmarshal(output, &result); err != nil {
+		return false, errors.Wrap(err, "failed to parse findmnt output")
+	}
+	l := len(result.Filesystems)
+	if result.Filesystems[l-1].Fstype != "fuse.zdbfs" {
```

Comment:
Shouldn't this list always be of len == 1 ?

Reply:
it's mounted twice, one bind mount for the child mounts to reflect on the host fs, and the second is the zdbfs mount. https://github.com/threefoldtech/zos/blob/07762a455f2a9a41b9cd8730463698312a9f22eb/pkg/qsfsd/qsfs.go#L146